src/protocol/OpenConnectionReply1.php

```diff
@@ -20,13 +20,15 @@ class OpenConnectionReply1 extends OfflineMessage{
 	public static $ID = MessageIdentifiers::ID_OPEN_CONNECTION_REPLY_1;
 
 	public int $serverID;
-	public bool $serverSecurity = false;
+	public bool $serverSecurity;
+	public int $cookie;
 	public int $mtuSize;
 
-	public static function create(int $serverId, bool $serverSecurity, int $mtuSize) : self{
+	public static function create(int $serverId, bool $serverSecurity, int $cookie, int $mtuSize) : self{
```

Comment:
Make cookie nullable by default to be in the BC

Reply:
Throw an exception if serverSecurity is set to true and no cookie is specified?

Reply:
> Throw an exception if serverSecurity is set to true and no cookie is specified?

is there such a possibility? should we add?

Reply:
For consistency, yes. We don't want to create a packet which have a cookie in their internal state without server security.